### PR TITLE
DATAGRAPH-752 - Add Spring Data Neo4j 4 Example to Spring-Data-Examples.

### PR DESCRIPTION
**Pull request by luanne**

### CODE REVIEW

File: neo4j/example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    <dependencies>
+
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-neo4j</artifactId>
+            <version>${spring-data-neo4j.version}</version>
+        </dependency>
+
+        <dependency>
```
Comment (olivergierke) [inline neo4j/example/pom.xml]: What's the reason we need these test JARs in the classpath? I would've expected to only see the Spring Data Neo4j dependency. Anything I am missing?

Comment (jexp) [inline neo4j/example/pom.xml]: They are needed for the in-memory server, and test-jar dependencies are not resolved transitively by maven :(

Comment (olivergierke) [inline neo4j/example/pom.xml]: But which of the two is it then? Does it make sense to make them a dedicated artifact? It just looks weird having to refer to something test related from a non-test scope.

Comment (luanne) [inline neo4j/example/pom.xml]: We need them all unfortunately. The spring-data-neo4j test-jar contains the InProcessServer and the others contain the test utilities and TestServer from neo4j-ogm

Comment (jexp) [inline neo4j/example/pom.xml]: Actually we can clean it up a bit, which I did here: https://github.com/spring-projects/spring-data-examples/pull/130

We can add an (JIRA) issue for a separate test module both in the `neo4j-ogm` and in `spring-data-neo4j-4`, should be pretty straightforward to add, not sure we can make it in time for 4.0.GA though?

**Comment by olivergierke** [comment]
Polished and merged (see #131).

**Pull request closed without merging**